core/templates/pages/exploration-player-page/services/extract-image-filenames-from-model.service.ts

```diff
@@ -191,26 +195,31 @@ export class ExtractImageFilenamesFromModelService {
     let collapsibleTagList = dummyDocument.getElementsByTagName(
       'oppia-noninteractive-collapsible');
     for (let i = 0; i < collapsibleTagList.length; i++) {
-      let contentWithValue = JSON.parse(
-        collapsibleTagList[i].getAttribute('content-with-value'));
-      let collapsibleDocument = (
-        new DOMParser().parseFromString(contentWithValue, 'text/html'));
-      imageTagLists.push(
-        collapsibleDocument.getElementsByTagName('oppia-noninteractive-image'));
+      let attribute = collapsibleTagList[i].getAttribute('content-with-value');
+      if (attribute !== null) {
+        let contentWithValue = JSON.parse(attribute);
+        let collapsibleDocument = (
+          new DOMParser().parseFromString(contentWithValue, 'text/html'));
+        imageTagLists.push(
+          collapsibleDocument.getElementsByTagName(
+            'oppia-noninteractive-image'));
+      }
     }
 
     // Add images that are embedded in tabs.
     let tabsTagList = dummyDocument.getElementsByTagName(
       'oppia-noninteractive-tabs');
     for (let i = 0; i < tabsTagList.length; i++) {
-      let contentsWithValue = JSON.parse(
-        tabsTagList[i].getAttribute('tab_contents-with-value'));
-      for (let contentWithValue of contentsWithValue) {
-        let tabDocument = (
-          new DOMParser().parseFromString(contentWithValue.content, 'text/html')
-        );
-        imageTagLists.push(
-          tabDocument.getElementsByTagName('oppia-noninteractive-image'));
+      let attribute = tabsTagList[i].getAttribute('tab_contents-with-value');
+      if (attribute !== null) {
```

Comment:
Why is this done? When can attribute be `null`?

Reply:
ditto below

Reply:
Created an attribute variable that gets the given attribute from a tab. If the given attribute does not exist, then attribute is null which means skip the current tab. If attribute is not null, then add images that are embedded in the given tab.

The same for the ones below:
Created an attribute variable that gets the given filename attribute from an imageTag. If the given filename does not exist, then filename is null which means skip the current imageTag. If filename is not null, then add the file to filenames.

Created an attribute variable that gets the given math-content attribute from a mathTag. If the given attribute does not exist, then attribute is null which means skip the current mathTag. If attribute is not null, then add the SVG filename to filenames.